<launch>
  <arg name="output_screen" default="true"/>
  <arg name="image_topic" default="image" />
  <arg name="image_depth_topic" default="image_depth" />
  <arg name="camera_info_topic" default="cam_info" />
  <arg name="debug_view" default="true"/>
  <arg name="marker_debug" default="true"/>
  <arg name="poly_array" default="polygon_array" />
  <arg name="frame_id" default="target_object"/>
  <arg name="camera" default="rs_d435"/>
  <arg name="gui" default="true"/>
    
  <arg name="INPUT_CLOUD" value="/rs_d435/depth_registered/points" />


  <node pkg="nodelet" type="nodelet" name="far_object_detection_nodelet" args="manager" output="screen" if="$(arg output_screen)"/>

  <node pkg="nodelet" type="nodelet" name="far_object_detection"
        args="load aerial_robot_perception/FarObjectDetection far_object_detection_nodelet" output="screen">


    <remap from="image" to="/rs_d435/color/image_rect_color" />
    <remap from="depth_image" to="/rs_d435/aligned_depth_to_color/image_raw" />
    <remap from="cam_info" to="/rs_d435/aligned_depth_to_color/camera_info" />
    
    <param name="debug_view" value="$(arg debug_view)"/>
    <param name="frame_id" value="$(arg frame_id)"/>
  </node>

</launch>
